### DISCUSSION TITLE
Adding Sustain

### DISCUSSION THREAD

**Discussion by boyberg** [bangcorrupt/hacktribe#84] [Ideas]
Out of all the annoying quirks of the electribe, this has annoyed me the most. Playing a note and it cutting off immediately after releasing a pad or not being able to sustain a note no matter how long you press (I think I have mostly resulted to volume modulation for attack).

Would it be possible to integrate at least a gate mode for the envelope to create an ASR envelope like on the minilogue or the old electribes? An adsr would be even better, I think that would be difficult though given the e2s controls...

**Comment by bangcorrupt** [comment]
I haven't looked into it yet, but I reckon the whole ADSR envelope function is there, just with fixed values for the unused parameters.  I may be way off, I might know more in a few months if I can reverse the comms between the CPU and DSP.

**Reply by juanjo1138** [reply]
Hi!

How it could work with a single button:

• The release control would still work with the 'Amp EG' button off.

• …or better, change the behavior of the Decay/Release control so that in the first half of its travel it works as Decay (without sustain) and in the second half as Release (from the moment the note is released).

**Reply by bangcorrupt** [reply]
Hello.  If I can get control of the UI some day we could use shift to have ADSR on 2 knobs.

**Reply by juanjo1138** [reply]
It could be fine, but I wouldn't complicate the workflow of the electribe; I would personally prefer this simple control that I have described above:

The first half of the knob would be for percussive sounds and the second for pads like.

Keep in mind that when the release is very long we are no longer in a percussive sound, and the length of the interpreter's touch is not relevant either, so the transition from one type of sound to another would be very natural.

**Reply by boyberg** [reply]
Awesome! I actually remember reading that someone discovered there is a value for release, it's just left blank

**Reply by ptowns** [reply]
Why is it labeled decay/release, when its actually gate/release! But this is one of my biggest frustrations with the electribe. The arp is almost pointless because of this. Even a ifx reverb would help this. Currently I deal with it by turning amp eg off sending arp synth to a mfx reverb to make it not cut so abruptly and feel like a bit more of a release. 
As a techno guy an arp that is focused on rythem and gate is awesome! Just kills me you have to pick between release or gates! Wtf! 
Little food for thought, the E2 sends midi note on off rapidly, regardless if you are still holding the pad. no hold or gate on the midi. Also someone told me that if you use a midi keyboard it will sustain on the amp eg mode. ?? Idk, thought I had it working once but then couldn't get to work another time, Not sure.

**Reply by bangcorrupt** [reply]
> Why is it labeled decay

With Amp EG off it is decay for the filter.  Separate full ADSR for amp and filter would really open things up.  

> pick between release or gates

With the factory synth firmware it doesn't even gate the PCM, just does nothing to the amp with Amp EG off.

> E2 sends midi note on off rapidly

In gate arp mode?  I get this at the bottom of the XY pad, moving to the top of the pad holds the gate open until the next note is triggered.  

> if you use a midi keyboard it will sustain on the amp eg mode

I tried, it doesn't work. You could probably send envelopes as CC to the part volume.  

We can use the envelopes in the Modulation section to have either attack or decay with sustain when Amp EG is off, but it doesn't help with release.  I like to use this to add a punchy spike, then slam into the new KP2 Compressor and resample.  I think the Punch IFX is actually part of the envelope function, it would be nice to separate it from the IFX slot.

**Comment by ptowns** [comment]
I was playing around with this last night. A few thoughts. Because gates seems useless when using amp EG, if the gate in the step edit / sequencer controlled gate AND release that would be better then nothing. Only would become a problem when using filter eg and no amp eg. 🤔

BUT if we could have gate as a modulation source  and release as a modulation destination !👌🏾

**Reply by bangcorrupt** [reply]
> release as a modulation destination

This is definitely possible, along with Attack, EG Int and Resonance. 

I haven't looked into modulation sources very far.  There are sine and unipolar square LFO's I'll enable at some point.  

I'm hoping I can combine the functionality of motion sequences, groove templates and the sequencer somehow to  make it more versatile.

**Reply by ptowns** [reply]
Have been fighting with this more and thinking about it. Obviously a single modulation slot is extremely limiting, BUT velocity and gate are stored values per step. Allowing them to be modulation sources, especially gate time, would allow us to turn amp eg on, meaning the gate value would do nothing, then as discussed earlier route it to release or another modulation destination and have like a extra sequencer lane. 
Not to go to far in the weeds here, but also each step stores 4 notes and if you put a voice to mono, there are 3 sequencer lanes that are ignored and sit there doing nothing?!? 'Seq 1, 2, 3, 4, Velocity,  and Seq G(gate)' as modulation sources would be huge!

**Reply by ptowns** [reply]
More exploration last night.  Something strange going on with the gate. So how come one shot samples ignore the gate in the sequencer but not the decay 'gate' knob???!?! Also whats with the strange resolution of gate times on both the knob and in the sequencer. Work around I have been using, not perfect, but set a one shot sound to loop mode, then it will fallow the gate. Still no release, but at least its a little extra sound shaping.  
In a dream world edit knob = start and loop points 0 to 100(or a single cycle), no reverse, and shift edit knob would be end point. Meaning no need for a gate knob.
